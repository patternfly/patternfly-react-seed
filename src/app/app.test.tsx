import React from 'react';
import { App } from '@app/index';
import { mount, shallow } from 'enzyme';
import { Button } from '@patternfly/react-core';

describe('App tests', () => {
  test('should render default App component', () => {
    const view = shallow(<App />);
    expect(view).toMatchSnapshot();
  });

  it('should render a nav-toggle button', () => {
    const wrapper = mount(<App />);
    const button = wrapper.find(Button);
    expect(button.exists()).toBe(true);
  });

  it('should hide the sidebar when clicking the nav-toggle button', () => {
    const wrapper = mount(<App />);
    const button = wrapper.find('#nav-toggle').hostNodes();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded'));
    button.simulate('click');
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed'));
  });
});
