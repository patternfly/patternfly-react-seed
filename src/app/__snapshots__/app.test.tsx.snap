// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<BrowserRouter>
  <AppLayout>
    <AppRoutes />
  </AppLayout>
</BrowserRouter>
`;
